oc_cdtapi/PgAPI.py

```
@@ -217,169 +245,226 @@ def post_historicaldelivery(self, request):
-            200
+            dict | None: The task dict, or ``None`` if not found.
         """
         try:
-            payload = {"id": task_id}
-            req = f"rest/api/1/tasks"
-            res = self.get(req, params=payload)
+            if self._api_version == "v2":
+                req = f"api/v2/tasks/{task_id}"
+                res = self.get(req)
+                return res.json().get("data")
+            else:
```

Ruff (0.15.15)

[warning] Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)